### PR TITLE
Alerting: Fix HCL export for alerts with non-zero "for" field.

### LINKED ISSUE #76734: Alerting: HCL export does not format for duration as expected
### What happened?

The `Export Rule Group` for Terraform (HCL) has invalid duration for the `for` string:
 
For example:
 ```
    no_data_state  = "NoData"
    exec_err_state = "Error"
    for            = 300
```

The expected format as written in the Terraform provider docs for Alert Groups: https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/rule_group indicate the `for` string needs the time measurement (`2m` in the example):

```
rule {
    name           = "My Alert Rule 1"
    for            = "2m"
    condition      = "B"
    no_data_state  = "NoData"
    exec_err_state = "Alerting"
```

### What did you expect to happen?

The exported HCL looks like:

```
    no_data_state  = "NoData"
    exec_err_state = "Error"
    for            = "2m"
```

### Did this work before?

Unknown

### How do we reproduce it?

1. Export an alert rule with a `for` field as HCL
2. Attempt to apply the exported definution

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: v10.2.0-61719
OS: N/A
Browser: N/A

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by stevesg**
The Terraform provider requires the `for` field as a duration string, not seconds. We also document it [here](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/rule_group#example-usage).

Rationale for the change:
- The HCL output appears to be special, in that it omits the field is it is zero.
- The HCL encoder does not appear to format the Prometheus Duration type correctly (`2m` becomes `120000000000`)

I'm not familiar with the HCL encoder, perhaps there is a nicer way to override the conversion for this field.

Fixes #76734

### CODE REVIEW

File: pkg/services/ngalert/api/compat.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func AlertRuleExportFromAlertRule(rule models.AlertRule) (definitions.AlertRuleE
 		IsPaused:     rule.IsPaused,
 	}
 	if rule.For.Seconds() > 0 {
-		result.ForSeconds = util.Pointer(int64(rule.For.Seconds()))
+		result.ForString = util.Pointer(model.Duration(rule.For).String())
```
Comment (grobinson-grafana) [inline pkg/services/ngalert/api/compat.go:189]: Why we need to turn a `time.Duration` into `model.Duration` before encoding it as a string? I'm not very familiar this with code, so there might be a good reason for it that I'm just not seeing.

File: pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go [hunk 2]
```diff
@@ -243,10 +243,13 @@ type AlertRuleExport struct {
 	NoDataState  NoDataState         `json:"noDataState" yaml:"noDataState" hcl:"no_data_state"`
 	ExecErrState ExecutionErrorState `json:"execErrState" yaml:"execErrState" hcl:"exec_err_state"`
 	For          model.Duration      `json:"for" yaml:"for"`
-	ForSeconds   *int64              `json:"-" yaml:"-" hcl:"for"`
-	Annotations  *map[string]string  `json:"annotations,omitempty" yaml:"annotations,omitempty" hcl:"annotations"`
-	Labels       *map[string]string  `json:"labels,omitempty" yaml:"labels,omitempty" hcl:"labels"`
-	IsPaused     bool                `json:"isPaused" yaml:"isPaused" hcl:"is_paused"`
+	// ForString is used to:
+	// - Only export the for field for HCL if it is non-zero.
```
Comment (grobinson-grafana) [inline pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go:247]: Zero should be formatted as "0s" no? I suppose it doesn't make a difference?

**Review by yuri-tceretian — APPROVED** [review]
LGTM

File: pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go (hunk 2 above)
Comment (grobinson-grafana) [reply pkg/services/ngalert/api/tooling/definitions/provisioning_alert_rules.go:247]: Good to know!

File: pkg/services/ngalert/api/compat.go (hunk 1 above)
Comment (stevesg) [reply pkg/services/ngalert/api/compat.go:189]: I don't have a good answer other than we use the Prometheus Duration parser in other places of the API (and the [Terraform Provider](https://github.com/grafana/terraform-provider-grafana/blob/494629dd0aeea2b6a9602da6b1bc87206ef4daa3/internal/common/schema.go#L70)), so I kept to the same type for consistency.

**Comment by stevesg** [comment]
Thanks for fixing the tests, forgot to run the whole suite.

File: pkg/services/ngalert/api/compat.go (hunk 1 above)
Comment (grobinson-grafana) [reply pkg/services/ngalert/api/compat.go:189]: Looks like it's quite different from `time.Duration` (https://github.com/prometheus/common/blob/main/model/time.go#L204) 😄

**Review by grobinson-grafana — APPROVED** [review]

**Pull request merged**